apps/xprof_gui/priv/src/components/monitoring/Graph/Graph.jsx

```diff
@@ -3,21 +3,55 @@ import React from 'react';
 import C3Chart from 'react-c3js';
 import 'c3/c3.css';
 import { AXIS, DATA, GRID, POINT, TRANSITION } from '../../../constants';
+import Grid from './Grid/Grid';
+
+function composeID(q) {
+  const modFuncArity = q.split(':');
+  const mod = modFuncArity[0];
+  const func = modFuncArity[1].split('/')[0];
+  const arity = modFuncArity[1].split('/')[1];
+  return `${mod}${func}${arity}`;
+}
 
 const propTypes = {
   dps: PropTypes.arrayOf(PropTypes.object).isRequired,
+  type: PropTypes.string.isRequired,
+  query: PropTypes.string.isRequired,
```

Comment:
If we are composing the ID inside `GraphPanel`, in the end we are only passing the ID to the `Graph` and the panel has access to the whole `m` object.
In case of renaming the JSON object I think this is a good idea, as for now the object named `mfa` has also a `mfa` property which is a little bit confusing (that's why I renamed it somewhere to `m` in loops or arguments). I propose we:
- rename the `mfas` to `monitoredCollection`
- rename the `mfa` to `monitored`.

What do you think?

Reply:
right, so `query` is only there so that the `Graph` can show it as a title. +1 for renamings